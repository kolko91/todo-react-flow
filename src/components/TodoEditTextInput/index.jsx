import React, { PureComponent } from 'react';
import styled from 'styled-components';

import type { Title } from 'types/todos';

const EditField = styled.input`
    display: block;
    width: 506px;
    padding: 13px 17px 12px 17px;
    margin: 0 0 0 43px;
`;

type Props = {
  onSave: (text: Title)=>void,
  title: Title
}

export default class TodoEditTextInput extends PureComponent<Props, {title: Title}> {
  state = {
    title: this.props.title,
  }

  handleSubmit = (e: SyntheticKeyboardEvent<HTMLButtonElement>) => {
    const { target, which } = e;
    if (target instanceof HTMLInputElement) {
      const text = target.value.trim();
      if (which === 13) {
        this.props.onSave(text);
      }
    }
  }

  handleChange = (e: Event) => {
    const { target } = e;
    if (target instanceof HTMLInputElement) {
      this.setState({ title: target.value });
    }
  }

  handleBlur = (e: Event) => {
    const { target } = e;
    if (target instanceof HTMLInputElement) {
      this.props.onSave(target.value);
    }
  }


  render() {
    return (
      <EditField
        type="text"
        autoFocus="true"
        value={this.state.title}
        onBlur={this.handleBlur}
        onChange={this.handleChange}
        onKeyDown={this.handleSubmit}
      />
    );
  }
}

