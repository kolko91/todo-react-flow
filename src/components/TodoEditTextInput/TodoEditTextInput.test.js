import React from 'react';
import renderer from 'react-test-renderer';
import TodoEditTextInput from './index';


describe('component', () => {
  describe('TodoEditTextInput', () => {
    it('should render correctly', () => {
      const props = {
        onSave: jest.fn(),
        title: 'Test',
      };
      const tree = renderer.create(<TodoEditTextInput {...props} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
