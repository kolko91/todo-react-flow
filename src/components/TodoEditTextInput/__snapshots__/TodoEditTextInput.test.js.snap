// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component TodoEditTextInput should render correctly 1`] = `
<input
  autoFocus="true"
  className="sc-bdVaJa fpNejE"
  onBlur={[Function]}
  onChange={[Function]}
  onKeyDown={[Function]}
  type="text"
  value="Test"
/>
`;
